src/extra/python/isca/codebase.py

```diff
@@ -217,7 +217,9 @@ def _log_line(self, line):
 
     @useworkdir
     @destructive
-    def compile(self, debug=False, optimisation=None):
+    # def compile(self, debug=False, optimisation=None):
```

Comment:
Please could you remove the changes for supporting source modifications for now?  I think it is a good idea, but we must be careful how we implement it so as not to break other people's existing experiment scripts.  We should open a separate issue to track and implement this.